flycheck.el

```diff
@@ -2114,7 +2156,7 @@ Return a list of `flycheck-verification-result' objects."
         (enabled (flycheck-checker-get checker 'enabled))
         (verify (flycheck-checker-get checker 'verify)))
     (when enabled
-      (let ((result (funcall enabled)))
+      (let ((result (flycheck-may-enable-checker checker)))
```

Comment:
Can you explain this part? (IIUC, this will return the same result, since we'll have reset the cache in the meantime?)

Reply:
`-verify-checker` ended up calling the `:enabled` predicate twice: once through `verify-generic-checker`, and once through `may-use-checker`.

Because `-may-enable` tests `:enabled` *and* caches its result, using `-may-enable` here ensures we run the predicate only once.

The `(funcall enabled)` above was obviously bypassing the cached result.

(Note that `-verify-checker` is seldom used though)